### PR TITLE
Notifications Manager: Revamp Mark IV

### LINKED ISSUE #4069: NotificationsManager Revamp
- [x] Rename the entire class: `PushNotificationsManager` (perhaps) would make more sense, since Notifications themselves are managed by Simperium.
- [x] The new NotificationsManager class should be implemented as a Singleton, instead of working as a collection of static methods. (Or perhaps as an instance maintained by the App Delegate?).
- [x] Encapsulate the Push Notification Handling code into several closures (or helper classes). (Candidate: `PushNotificationHandler`).

### LINKED ISSUE #4454: Notifications: FetchCompletionHandler may never get called
Props to @astralbodies for reporting this one (Thank you Aaron!)

```
WordPress[680:653930] Warning: Application delegate received call to -application:didReceiveRemoteNotification:fetchCompletionHandler: but the completion handler was never called.```
```

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Details:
In this PR we're (finally) removing `NotificationsManager` class, and replacing it with new Swift code (plus unit tests). Testing scenarios outlined below!

Closes #4069
Closes #4454

--

#### Scenario 1: Push Authentication in BG
1. Log into WPiOS
2. Send the app to Background
3. In a desktop browser, open *WordPress.com* and log into an account that requires 2FA
4. Wait for the Push Authentication Notification to arrive and tap on it

As a result, WPiOS should display an Alert indicating that there's a login attempt on the web, requesting the user to either Accept / Ignore.

--

#### Scenario 2: Push Authentication in FG
1. Log into WPiOS and keep the app in foreground
2. In a desktop browser, open *WordPress.com* and log into an account that requires 2FA

As a result, WPiOS should display an Alert indicating that there's a login attempt on the web, requesting the user to either Accept / Ignore.

--

#### Scenario 3: Comment Notifications
1. Log into WPiOS and keep the app in background / killed
2. Receive a comment in any of your blogs
3. Wait until the Push Notification arrives and tap on it

As a result, WPiOS should launch, and the new comment notification should be onscreen.

Needs Review: @diegoreymendez 
Thanks in advance Diego!

### CODE REVIEW

**Comment by astralbodies** [comment]
@diegoreymendez Are you able to review this soon?

**Comment by diegoreymendez** [comment]
@astralbodies - Yes, will do.

File: WordPress/Classes/Utility/PushNotificationsManager.swift [hunk 1]
```diff
@@ -20,42 +21,77 @@ final public class PushNotificationsManager : NSObject
     
     /// Returns the shared PushNotificationsManager instance.
     ///
-    static let sharedInstance = PushNotificationsManager()
+    public static let sharedInstance = PushNotificationsManager()
```
Comment (diegoreymendez) [inline WordPress/Classes/Utility/PushNotificationsManager.swift]: This class, and thus the methods, shouldn't need to be public.

Feel free to skip in this comment though, as the visibility of this class doesn't fall inside the scope of this PR.

**Comment by diegoreymendez** [comment]
All test cases are working flawlessly here.

File: WordPress/Classes/Utility/PushNotificationsManager.swift (hunk 1 above)
Comment (diegoreymendez) [inline WordPress/Classes/Utility/PushNotificationsManager.swift]: Actually it seems like you've added `public` in several spots.  Is there any particular reason?

**Comment by diegoreymendez** [comment]
Hey @jleandroperez!  The test cases work perfectly.  I just added a comment in the code for you to evaluate.

Also, when compiling the unit tests I'm getting this error, which I believe may be fixed by merging the latest from develop:

```
ld: '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks/XCTest.framework/XCTest' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by jleandroperez** [comment]
@diegoreymendez thanks for the review!!

- *public* keyword removed. I believe that was added along the way, while i was experimenting with ObjC / Swift unit tests, and had to patch it all.
- Linker error is not showing up around here (latest develop has been merged)

May i bother you with a quick peek?. Thanks!!

**Comment by diegoreymendez** [comment]
:shipit:

**Comment by jleandroperez** [comment]
Thank you sir!

**Pull request merged**